### PR TITLE
Add algorithm to compute real Gaunt coefficient

### PR DESCRIPTION
**Pull request by AntiPhoton47**
I have coded a symbolic algorithm to compute the integral of three real spherical harmonics based on the outline presented in Homeier and Steinborn (1996). The appropriate import lines, references, and documentation have also been added throughout the entire file.

#### References to other Issues or PRs

I proposed these changes with the help of @smichr in pull request #24035, but I am new to GitHub and didnt merge the changes properly so I am opening a new pull request.

#### Brief description of what is fixed or changed

An algorithm to compute the real Gaunt coefficient has been added as a function titled real_gaunt, along with the appropriate import lines, references, and documentation scattered throughout the wigner.py file. A test function titled test_realgaunt was added to the testing file within the physics module as well. Lastly, the test functions for the clebsch-gordon function were all condensed down to one.

#### Other comments

The algorithm uses the already included Gaunt algorithm, but I can find another if this is problematic (due to copyright issues perhaps). Please also let me know if anyone finds any bugs or other conflicting lines.

EDIT: Very big thanks to Christopher Smith for suggesting a number of modifications that have made the program more concise and reduced the runtime significantly.

#### Release Notes

* physics.wigner
  * Added `real_gaunt` function to compute the integral of 3 real spherical harmonics.

### CODE REVIEW

File: sympy/physics/wigner.py [hunk 1]
```diff
@@ -715,44 +719,194 @@ def gaunt(l_1, l_2, l_3, m_1, m_2, m_3, prec=None):
...
     if (m_1 + m_2 + m_3) != 0:
         return S.Zero
     if (abs(m_1) > l_1) or (abs(m_2) > l_2) or (abs(m_3) > l_3):
         return S.Zero
+    bigL, remL = divmod(l_1 + l_2 + l_3, 2)
+    if remL % 2:
+        return S.Zero
 
     imin = max(-l_3 + l_1 + m_2, -l_3 + l_2 - m_1, 0)
     imax = min(l_2 + m_2, l_1 - m_1, l_1 + l_2 - l_3)
```
Comment (smichr) [inline sympy/physics/wigner.py:731]:
```suggestion
    imax = min(l_2 + m_2, l_1 - m_1, l_1 + l_2 - l_3)

```

File: sympy/physics/wigner.py [hunk 2]
```diff
@@ -715,44 +719,194 @@ def gaunt(l_1, l_2, l_3, m_1, m_2, m_3, prec=None):
...
         _Factlist[2 * bigL + 1]/ \
         (_Factlist[bigL - l_1] *
          _Factlist[bigL - l_2] * _Factlist[bigL - l_3])
-
     sumres = 0
     for ii in range(int(imin), int(imax) + 1):
         den = _Factlist[ii] * _Factlist[ii + l_3 - l_1 - m_2] * \
             _Factlist[l_2 + m_2 - ii] * _Factlist[l_1 - ii - m_1] * \
             _Factlist[ii + l_3 - l_2 + m_1] * _Factlist[l_1 + l_2 - l_3 - ii]
         sumres = sumres + Integer((-1) ** ii) / den
```
Comment (smichr) [inline sympy/physics/wigner.py:749]:
```suggestion
        sumres = sumres + Integer((-1) ** ii) / den

```

File: sympy/physics/wigner.py [hunk 3]
```diff
@@ -715,44 +719,194 @@ def gaunt(l_1, l_2, l_3, m_1, m_2, m_3, prec=None):
...
         _Factlist[l_1 - m_1] * _Factlist[l_1 + m_1] * _Factlist[l_2 - m_2] * \
         _Factlist[l_2 + m_2] * _Factlist[l_3 - m_3] * _Factlist[l_3 + m_3] / \
         (4*pi)
     ressqrt = sqrt(argsqrt)
-
     prefac = Integer(_Factlist[bigL] * _Factlist[l_2 - l_1 + l_3] *
                      _Factlist[l_1 - l_2 + l_3] * _Factlist[l_1 + l_2 - l_3])/ \
         _Factlist[2 * bigL + 1]/ \
         (_Factlist[bigL - l_1] *
          _Factlist[bigL - l_2] * _Factlist[bigL - l_3])
```
Comment (smichr) [inline sympy/physics/wigner.py:743]:
```suggestion
         _Factlist[bigL - l_2] * _Factlist[bigL - l_3])

```

File: sympy/physics/wigner.py [hunk 4]
```diff
@@ -715,44 +719,194 @@ def gaunt(l_1, l_2, l_3, m_1, m_2, m_3, prec=None):
...
     imax = min(l_2 + m_2, l_1 - m_1, l_1 + l_2 - l_3)
-
     maxfact = max(l_1 + l_2 + l_3 + 1, imax + 1)
     _calc_factlist(maxfact)
-
     argsqrt = (2 * l_1 + 1) * (2 * l_2 + 1) * (2 * l_3 + 1) * \
         _Factlist[l_1 - m_1] * _Factlist[l_1 + m_1] * _Factlist[l_2 - m_2] * \
         _Factlist[l_2 + m_2] * _Factlist[l_3 - m_3] * _Factlist[l_3 + m_3] / \
         (4*pi)
     ressqrt = sqrt(argsqrt)
```
Comment (smichr) [inline sympy/physics/wigner.py:738]:
```suggestion
    ressqrt = sqrt(argsqrt)

```

File: sympy/physics/wigner.py [hunk 5]
```diff
@@ -715,44 +719,194 @@ def gaunt(l_1, l_2, l_3, m_1, m_2, m_3, prec=None):
...
         return S.Zero
+    bigL, remL = divmod(l_1 + l_2 + l_3, 2)
+    if remL % 2:
+        return S.Zero
 
     imin = max(-l_3 + l_1 + m_2, -l_3 + l_2 - m_1, 0)
     imax = min(l_2 + m_2, l_1 - m_1, l_1 + l_2 - l_3)
-
     maxfact = max(l_1 + l_2 + l_3 + 1, imax + 1)
     _calc_factlist(maxfact)
```
Comment (smichr) [inline sympy/physics/wigner.py:733]:
```suggestion
    _calc_factlist(maxfact)

```

File: sympy/physics/wigner.py [hunk 6]
```diff
@@ -699,13 +707,9 @@ def gaunt(l_1, l_2, l_3, m_1, m_2, m_3, prec=None):
 
     Jens Rasch (2009-03-24): initial version for Sage.
     """
-    if int(l_1) != l_1 or int(l_2) != l_2 or int(l_3) != l_3:
-        raise ValueError("l values must be integer")
-    if int(m_1) != m_1 or int(m_2) != m_2 or int(m_3) != m_3:
-        raise ValueError("m values must be integer")
+    lm = l_1, l_2, l_3, m_1, m_2, m_3 = [
```
Comment (smichr) [inline sympy/physics/wigner.py:710]:
```suggestion
    l_1, l_2, l_3, m_1, m_2, m_3 = [
```

**Comment by smichr** [comment]
If, in this branch, you `git reset head~8` and commit as "review edits" (and even squash that 4th commit into your 3rd commit) and then push here with `git push -f AntiPhoton47 patch-1` that will clean up the several small additions that we have made.

**Comment by AntiPhoton47** [comment]
I'm still doing something wrong and i'm not sure what it is. I am in my local directory and I first entered `git reset head~8`, then I entered `git commit -m "review edits"`, and finally I entered `git push -f origin patch-1`.

**Comment by AntiPhoton47** [comment]
The odd part is that my local clone is the version we had before I reset the head, but the version that was pushed to my remote repository is the original version I opened the pull request with.

**Comment by smichr** [comment]
Looks good to me. This will merge when tests pass.

**Pull request merged**